kafka-client/src/main/java/dev/responsive/kafka/internal/db/rs3/RS3WindowTable.java

```diff
@@ -0,0 +1,226 @@
+package dev.responsive.kafka.internal.db.rs3;
+
+import dev.responsive.kafka.internal.db.RemoteWindowTable;
+import dev.responsive.kafka.internal.db.WindowFlushManager;
+import dev.responsive.kafka.internal.db.rs3.client.LssId;
+import dev.responsive.kafka.internal.db.rs3.client.LssMetadata;
+import dev.responsive.kafka.internal.db.rs3.client.MeteredRS3Client;
+import dev.responsive.kafka.internal.db.rs3.client.Put;
+import dev.responsive.kafka.internal.db.rs3.client.RS3Client;
+import dev.responsive.kafka.internal.db.rs3.client.RS3ClientUtil;
+import dev.responsive.kafka.internal.db.rs3.client.WalEntry;
+import dev.responsive.kafka.internal.metrics.ResponsiveMetrics;
+import dev.responsive.kafka.internal.utils.WindowedKey;
+import java.util.Objects;
+import java.util.UUID;
+import org.apache.kafka.common.utils.Bytes;
+import org.apache.kafka.streams.state.KeyValueIterator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RS3WindowTable implements RemoteWindowTable<WalEntry> {
+  private static final Logger LOG = LoggerFactory.getLogger(RS3WindowTable.class);
+
+  private final String name;
+  private final UUID storeId;
+  private final RS3Client rs3Client;
+  private final RS3ClientUtil rs3ClientUtil;
+  private final PssPartitioner pssPartitioner;
+  private final RS3WindowedKeySerde keySerde = new RS3WindowedKeySerde();
+
+  // Initialized in `init()`
+  private LssId lssId;
+  private Long fetchOffset;
+  private RS3WindowFlushManager flushManager;
+
+  public RS3WindowTable(
+      final String name,
+      final UUID storeId,
+      final RS3Client rs3Client,
+      final PssPartitioner pssPartitioner,
+      final ResponsiveMetrics responsiveMetrics,
+      final ResponsiveMetrics.MetricScopeBuilder scopeBuilder
+  ) {
+    this(
+        name,
+        storeId,
+        new MeteredRS3Client(
+            Objects.requireNonNull(rs3Client),
+            Objects.requireNonNull(responsiveMetrics),
+            Objects.requireNonNull(scopeBuilder)
+        ),
+        pssPartitioner
+    );
+  }
+
+  public RS3WindowTable(
+      final String name,
+      final UUID storeId,
+      final RS3Client rs3Client,
+      final PssPartitioner pssPartitioner
+  ) {
+    this.name = Objects.requireNonNull(name);
+    this.storeId = Objects.requireNonNull(storeId);
+    this.rs3Client = Objects.requireNonNull(rs3Client); // TODO: Use metered client
```

Comment:
just want to make sure this doesn't get lost, is this planned for a followup PR?

Reply:
Oh actually I implemented this through the alternate constructor above. I'll remove the TODO.